src/main/java/service/impl/RemoveOperation.java

```diff
@@ -0,0 +1,19 @@
+package service.impl;
+
+import db.Storage;
+import model.Fruit;
+import model.dto.FruitRecordDto;
+import service.FruitOperationsService;
+
+public class RemoveOperation implements FruitOperationsService {
+    @Override
+    public int apply(FruitRecordDto fruitRecordDto) {
+        if (Storage.fruits
+                .get(new Fruit(fruitRecordDto.getFruitName())) < fruitRecordDto.getQuantity()) {
+            throw new RuntimeException(fruitRecordDto.getFruitName() + " out of Stock!");
+        }
+        fruitRecordDto.setQuantity(-fruitRecordDto.getQuantity());
+        FruitOperationsService fruitOperationsService = new AddOperation();
+        return fruitOperationsService.apply(fruitRecordDto);
```

Comment:
it is not readable. It took me some time to understand and you are changing you dto just subtract and save here. The number of lines will be the same but it will be more readable